File: android/src/main/java/com/lightningkite/khrysalis/time/TimeInterval.actual.kt
package com.lightningkite.khrysalis.time



inline class TimeInterval(val milliseconds: Long) {
    val seconds: Double get() = milliseconds / 1000.0
    val minutes: Double get() = seconds / 60.0
    val hours: Double get() = minutes / 60.0
    val days: Double get() = hours / 24.0
    val approximateYears: Double get() = days / 365.25
}

fun Long.milliseconds(): TimeInterval =
    TimeInterval(this.toLong())
fun Int.milliseconds(): TimeInterval =
    TimeInterval(this.toLong())
fun Int.seconds(): TimeInterval =
    TimeInterval(this.toLong() * 1000L)
fun Int.minutes(): TimeInterval =
    TimeInterval(this.toLong() * 60L * 1000L)
fun Int.hours(): TimeInterval =
    TimeInterval(this.toLong() * 60L * 60L * 1000L)
fun Int.days(): TimeInterval =
    TimeInterval(this.toLong() * 24L * 60L * 60L * 1000L)

File: ios/Khrysalis/src/time/TimeInterval.actual.swift
//Stub file made with Khrysalis 2 (by Lightning Kite)
import Foundation


//--- TimeInterval.{
//--- TimeInterval.Primary Constructor
//--- TimeInterval.milliseconds
public extension TimeInterval {
    var milliseconds: Int64 {
        return Int64(self * 1000)
    }
}
//--- TimeInterval.seconds
public extension TimeInterval {
    var seconds: Double {
        return self
    }
}
//--- TimeInterval.}

//--- Int.milliseconds()
//--- Int.seconds()
//--- Int.minutes()
//--- Int.hours()
//--- Int.days()
public extension Int {
    func milliseconds() -> TimeInterval { return TimeInterval(self) / 1000 }
    func seconds() -> TimeInterval { return TimeInterval(self) }
    func minutes() -> TimeInterval { return TimeInterval(self * 60) }
    func hours() -> TimeInterval { return TimeInterval(self * 60 * 60) }
    func days() -> TimeInterval { return TimeInterval(self * 60 * 60 * 24) }
}
